### PR TITLE
ci: speedup the CLA check

### PR DESCRIPTION
**Pull request by sergiusens**
Verify an email only once and don't verify commits that are already in master

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

### CODE REVIEW

**Comment by elopio** [comment]
Thank you for this!

It is failing with: error: unknown option `format="%(refname)"'

File: tools/cla_check.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+# XXX copied from https://github.com/kyrofa/cla-check
+import collections
+import os
+import subprocess
+import sys
+
+Commit = collections.namedtuple('Commit', ['email', 'hash'])
```
Comment (elopio) [inline tools/cla_check.py]: This should go after all the imports.

**Comment by sergiusens** [comment]
On mié, 23 ago, 2017 at 2:45 AM, Leo Arias <notifications@github.com> 
wrote:
> Thank you for this!
> 
> It is failing with: error: unknown option `format="%(refname)"'
> 

Yeah, seems the usable git on that docker image is not the latest and 
greatest :-)
>

File: tools/cla_check.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    split_output = (i.split('|') for i in output.split('\n') if i != '')
+    commits = [Commit(email=i[0], hash=i[1]) for i in split_output]
+    return commits
+
+
+def get_emails_from_commits(commits):
+    emails = set()
+    for c in commits:
+        output = check_output(['git', 'branch', '--contains', c.hash])
+        if '* master' not in output.split('\n'):
```
Comment (kyrofa) [inline tools/cla_check.py]: This of course means that the user is _on_ master. Is that really a requirement?

Comment (sergiusens) [inline tools/cla_check.py]: no, not the user, the commit

Comment (kyrofa) [inline tools/cla_check.py]: The output of this commit is a list of branches containing the commit in question. An asterisk is next to the branch the user is currently on.

Comment (sergiusens) [inline tools/cla_check.py]: oh ic `*`

**Pull request merged**